security/pkg/nodeagent/sds/server.go

```diff
@@ -101,6 +101,14 @@ func (s *Server) initDiscoveryService(options *Options, st SecretManager) error
 		return err
 	}
 
+	if _, err := os.Stat(options.UDSPath); err == nil {
+		log.Debugf("%q exists", options.UDSPath)
+
+		if err := os.Chmod(options.UDSPath, 0666); err != nil {
+			log.Errorf("failed to update %q permission", options.UDSPath)
+		}
```

Comment:
Should we fail fast and exit the program instead of fail silently? based on the motivation of the PR, if this `chmod` fails, the following istio proxy can't work?

Reply:
make sense, follow up - https://github.com/istio/istio/pull/7533